### PR TITLE
Allow Cloud machines to be reordered within pinned sections

### LINKED ISSUE #13086: Cloud machines: drag to reorder while preserving pinned sections
Allow dragging Cloud VM machine rows to choose their order, like workspace reordering, while preserving machine pinning rules.

User request verbatim: “just like how we are able to rearrange workspaes, we should be able to rearrange the vm machines and order, (obv preserve the pinning logic similar to pinning workspaces and pinning vms)”

Screenshot: the Cloud list shows two machines, pinned early-plum-alpaca followed by calm-cobalt-bison. This request concerns reordering the top-level machine rows, not remote workspaces or terminals within a machine.

Expected behavior:
- Drag machine rows to reorder them, with an insertion indicator consistent with workspace reordering.
- Keep pinned machines first and allow ordering within both the pinned and unpinned sections. Reordering must preserve existing pin membership and pin/unpin semantics, including sensible behavior at the pin boundary.
- Persist the order using the existing account/team-scoped Cloud machine pin/order store. Refreshing, reconnecting, collapsing/expanding rows, and app relaunch must retain the chosen order.
- New machines, removed machines, partial fleet/catalog updates, and account switches must reconcile safely without scrambling retained order or leaking order/pins between scopes.
- Reorder machine identities only: do not move remote workspaces/terminals between machines, recreate connections, or change machine routing IDs. Preserve selection, expansion state, and current pin icons.
- Preserve the special local-machine/pending-create rows if present; drops on child rows must not become ambiguous reparent operations.

Code leads on the reported main build: CloudMachinePinStore in Packages/macOS/CmuxCloudMachines already remembers stable order and partitions pinned/unpinned machines. CloudTreeOutlineView+Organization and CloudSidebarOrganizationDrop implement sibling reordering for descendants; extend the right owner for top-level machines instead of adding a competing order store. Inspect existing left-sidebar workspace drag/pin behavior as the interaction reference.

Validation: focused order/pin persistence and reconciliation tests, plus real drag behavior evidence when an isolated verification surface is available. No fake utilization or unrelated icon/name changes; those are tracked separately in #13071, #13072, and #13074.

### PR DESCRIPTION
**Pull request by austinywang**
Cloud machine headers currently cannot be dragged and their scoped pin/order store has no manual move operation. This adds top-level machine reordering with the existing workspace insertion line, preserving pinned-first ordering and machine identities.

- Drag headers or hover expanded children to insert at a machine boundary. Crossing the pin boundary clamps to the source tier; dragging never pins or unpins. This Mac and pending creates remain fixed.
- Persist moves through CloudMachinePinStore. Current membership, pin state, account scope, and account generation are checked before commit; partial snapshots retain hidden order, while complete reconciliation prunes deleted identities.
- Share moves across the machine context menu, accessibility actions, and the existing configured workspace-reorder keys while a machine header has focus. Descendant order, routing IDs, connections, selection, expansion, and current pin icons remain intact.

Validation: test-only header-drag commit followed by implementation; focused package tests cover order, pins, restart, account isolation, partial discovery and deletion. Native outline tests cover collapsed/expanded headers, insertion geometry, child hovering, cancellation, live refresh, stale drops, special/adopted rows and input parity. The native suites have an explicit non-tolerant hosted CI step. Local checks passed for project/test wiring, Swift file lengths, localization parity (all nine supported locales), and CI routing.

Trade-offs and limits: expanded-child hover resolves to the machine's trailing edge. The existing container type was extracted to keep CloudTreeOutlineView within its file budget. No Mac build or live GUI drag recording was requested/run; the cloud-Mac verification provisioner has not migrated to controller scheduling. Hosted Swift tests and structured review are pending. Final build tag: 13086-cloud-machine-ordering.

The scopes of #13081 (spacing) and #13082 (friendly names) were inspected; their branches were not copied. This change does not alter machine names, icons, spacing or utilization.

Fixes #13086

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/13090?ref=codesmith_pr_footer"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1792454333&installation_model_id=21920&pr_number=13090&ref=codesmith_pr_footer&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F13090&signature=87b93f09b2aa789ad31c8e8ca41e276215a0cbaf965247a4526a773b1df4e4ff"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

---
## Summary by cubic
Fixes #13086 by allowing Cloud machine headers to be reordered within their pinned or unpinned sidebar group. Previously headers could not be dragged; moves now persist per account scope without changing pin membership or projecting panes, and completing a header drag no longer raises an AppKit exception.

**Behavior**
- Drag a header or hover an expanded child to place it at a machine boundary; crossing tiers clamps to the source tier.
- Context-menu, keyboard, and accessibility actions share the same reorder path; Cloud focus consumes configured reorder keys for child or empty selections, while text fields keep existing routing.
- Moves use machine identities, preserve hidden order through partial refreshes, prune deleted identities during full reconciliation, and keep This Mac, pending creates, selection, expansion, children, and routing IDs intact.
- Pin state also survives on catalog-only cloud rows.

**Validation**
- Adds package and app-host coverage for persistence, account isolation, refreshes, stale drops, pin boundaries, drag behavior, rendered Observation updates, and command parity.
- CI runs the two Cloud ordering suites as separate focused invocations, fails on an empty selector, and still executes them when unrelated tests fail.

<sup>Written for commit 98eaeefc5a275cc093199b5d9517d9d67ed6e7e7. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/13090?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added persistent cloud machine reordering in the sidebar.
  * Reorder machines using drag-and-drop, keyboard shortcuts, context menus, and accessibility actions.
  * Preserved custom order across refreshes, relaunches, partial discovery, and pin changes.
  * Added safeguards against stale, invalid, cross-account, and unsupported reorder operations.

* **Documentation**
  * Updated cloud machine ordering guidance and examples.

* **Tests**
  * Added comprehensive coverage for reordering, persistence, accessibility, and edge cases.

### CODE REVIEW

File: Sources/Cloud/Sidebar/CloudTreeOutlineView+MachineOrdering.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        applyMachineOrder(machines)
+        return true
+    }
+
+    /// A boundary move is consumed even when it cannot change order, so a
+    /// focused Cloud header never accidentally reorders the local workspace.
+    func moveSelectedMachine(by offset: Int) -> Bool {
+        guard let outlineView,
+              let node = outlineView.item(atRow: outlineView.selectedRow) as? CloudTreeNode,
+              let id = node.machineOrderID else { return false }
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/Cloud/Sidebar/CloudTreeOutlineView+MachineOrdering.swift]: Fixed in 7fc97717ee. A focused Cloud outline now consumes reorder keys for descendant/local rows and empty selection as well as headers, so the shortcut cannot fall through to local workspace movement. Added focused acceptance assertions for these selections; hosted validation is pending.

*— Claude Code*

File: Sources/Cloud/Sidebar/CloudTreeOutlineView+MachineOrdering.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    func configureMachineReorderAccessibility(_ cell: NSView, node: CloudTreeNode) {
+        guard let id = node.machineOrderID, let actions = machineActions.ordering else {
+            cell.setAccessibilityCustomActions(nil)
+            return
+        }
+        cell.setAccessibilityCustomActions(machineMoveOptions.map { title, move in
+            NSAccessibilityCustomAction(name: title, handler: { [weak self] in
+                self?.moveMachine(id, move: move, using: actions) == true
+            })
+        })
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/Cloud/Sidebar/CloudTreeOutlineView+MachineOrdering.swift]: Fixed in 7fc97717ee. Accessibility queries filter actions through the same canMove predicate as the context menu. Availability is evaluated when AX requests it, so pinning a neighboring machine cannot leave stale actions exposed on an unpainted cell. Added a pin/unpin-without-repaint acceptance case; hosted validation is pending.

*— Claude Code*

File: Sources/Cloud/MachinesPanelViewModel+MachinePins.swift [hunk 3]
```diff
@@ -8,27 +8,62 @@ extension MachinesPanelViewModel {
...
+        }
+        actions.ordering = CloudMachineOrderingActions(
+            canMove: canMove,
+            move: { [weak self, weak machinePinStore] id, move in
+                guard canMove(id, move), let self, let machinePinStore,
+                      machinePinStore.move(move, machineID: id, machineIDs: self.currentMachineOrderIDs) else { return nil }
+                // Catalog reads are frozen while dragging; membership validation
+                // must still use the latest scoped catalog, never frozen rows.
+                return self.orderedMachines(
+                    MachineSnapshotBuilder.includingCatalogMachines(self.machines, catalog: self.scopedCatalogSnapshot())
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/Cloud/MachinesPanelViewModel+MachinePins.swift:53]: Investigated against Apple’s incremental Observation guidance: SwiftUI supports mixed ObservableObject/StateObject and @Observable model types. MachinesPanelView builds sidebarMachines during body evaluation; that getter reads the @Observable CloudMachinePinStore state, so both panels register that dependency directly. dc720deabd adds both a two-reader Observation test and two real NSHostingView outlines that must update from the shared store without objectWillChange or rootView replacement. The canonical review accepted this ownership boundary and is clean. The native hosted proof is still pending CI, so I am leaving this thread open until that proof completes. Reference: https://developer.apple.com/documentation/swiftui/migrating-from-the-observable-object-protocol-to-the-observable-macro

*— Claude Code*

File: Sources/Cloud/Sidebar/CloudTreeOutlineView+MachineOrdering.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/Cloud/Sidebar/CloudTreeOutlineView+MachineOrdering.swift]: Verified by hosted CloudMachineOrderingTests.commandParity: descendant/local/empty Cloud selections consume reorder keys without moving local workspaces. Focused run 35485862493 passed all 12 tests across both Cloud suites.

*— Claude Code*

File: Sources/Cloud/Sidebar/CloudTreeOutlineView+MachineOrdering.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/Cloud/Sidebar/CloudTreeOutlineView+MachineOrdering.swift]: Verified by the hosted accessibilityAvailability test: action availability follows live pin boundaries without repainting peer cells. Focused run 35485862493 passed.

*— Claude Code*

File: Sources/Cloud/MachinesPanelViewModel+MachinePins.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/Cloud/MachinesPanelViewModel+MachinePins.swift:53]: Rejected as a false positive after direct hosted proof. Both the shared Observation test and two real NSHostingView outlines passed in run 35485862493. They update from the one @Observable pin/order store without an ObservableObject broadcast or replacing rootView. The existing body read tracks the nested store, consistent with Apple’s incremental Observation documentation.

*— Claude Code*

File: cmuxTests/CloudMachineOrderingTests.swift [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+        let outlines = try hosts.map { try #require(findOutline(in: $0)) }
+        let coordinators = try outlines.map { try #require($0.delegate as? CloudTreeOutlineView.Coordinator) }
+        #expect(coordinators.allSatisfy { $0.nodes.compactMap(\.machineOrderID) == ["a", "b", "c", "d"] })
+        let actions = try #require(fixture.coordinator.machineActions.ordering)
+        #expect(actions.move("d", .top) != nil)
+        // Drain the scheduled UI transaction, then lay out without replacing
+        // rootView, manually invalidating the host, or publishing the adapter.
+        await withCheckedContinuation { continuation in
+            RunLoop.main.perform(inModes: [.common]) { continuation.resume() }
+        }
```
[the remark this answers is not in the document]
Comment (austinywang) [reply cmuxTests/CloudMachineOrderingTests.swift]: Fixed in 47c5bd79ef: the test now waits on both coordinators reaching the expected rendered order, allowing multiple main-loop transactions until a five-second deadline. The final concrete order assertion remains. Canonical review of this exact commit is clean.

*— Claude Code*

File: cmuxTests/CloudMachineOrderingTests.swift [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+        second.localWorkspacesProvider = { [] }
+        second.readCatalog()
+        let actions = try #require(fixture.coordinator.machineActions.ordering)
+        await confirmation("Both order readers invalidate", expectedCount: 2) { changed in
+            withObservationTracking { _ = fixture.model.sidebarMachines } onChange: { changed() }
+            withObservationTracking { _ = second.sidebarMachines } onChange: { changed() }
+            #expect(actions.move("d", .top) != nil)
+        }
+        #expect(fixture.model.sidebarMachines.map(\.id) == ["d", "a", "b", "c"])
+        #expect(second.sidebarMachines.map(\.id) == ["d", "a", "b", "c"])
```
[the remark this answers is not in the document]
Comment (austinywang) [reply cmuxTests/CloudMachineOrderingTests.swift:312]: The hosted proof was restored in 05c939005f after the canceled job logs identified the actual stall: super.draggingEnded invoked an unimplemented optional AppKit callback during the first header test, before the hosted proof began. The production callback was fixed, and the restored two-panel test passed in focused run 35485862493.

*— Claude Code*
